workflows/codefresh-csdp/versions/1.1.5/images/report-image-info/src/index.js

```diff
@@ -101,11 +152,70 @@ async function main() {
     }`;
 
     const registryResult = await graphQLClient.request(registryMutation, imageRegistryVars)
-        .catch(_handleQlError('failed to create image registry'));
+      .catch(_handleQlError('failed to create image registry'));
 
     console.log(JSON.stringify(registryResult, null, 2));
 
-    console.log('image reported successfully');
+    return registryResult;
+}
+
+/**
+ * @param history {[*]}
+ * @returns {[{size, created: (string|Boolean), instruction: string, args: string}]}
+ */
+function _getLayersFromHistory(history) {
+
+    let shell             = '/bin/sh -c';
+    let instructionPrefix = '#(nop)';
+
+    history = history.reverse().map((row) => {
+        row = row || {};
+
+        const CreatedBy = _.get(row, 'created_by', '');
+        let instruction, args;
+
+        if (CreatedBy.indexOf(instructionPrefix) === -1) {
+            instruction = COMMON_INSTRUCTIONS.run;
+            args        = CreatedBy.replace(shell, '').trim();
+        } else {
+            let cmd               = CreatedBy.substr(CreatedBy.indexOf(instructionPrefix) +
+              instructionPrefix.length).trim();
+            let nextSpacePosition = cmd.indexOf(' ');
+
+            instruction = cmd.substr(0, nextSpacePosition).trim();
+            args        = cmd.substr(nextSpacePosition).trim();
+
+            if (instruction === COMMON_INSTRUCTIONS.shell) {
+                shell = (args[0] === '[' && args[args.length - 1] === ']') ? (args
+                  .slice(1, -1)
+                  .replace(/[\"\']/g, '')
+                  .split(',')
+                  .map(part => {return part.trim();})
+                  .join(' ')) : args;
+            }
+        }
+
+        return {
+            created: row.created,
+            size: 0,
+            instruction,
+            args
+        };
+    }).reverse();
+
+    return history;
+}
+
+async function _buildDockerfile() {
+    try {
+        if (inputs.dockerfileContent) {
+            return inputs.dockerfileContent
+        }
+        return await fs.readFile('/tmp/Dockerfile')
```

Comment:
Let's make the path to dockerfile an env variable as well. It won't be used by codefresh-report-image plugin, but can be useful when using just images or workflow templates.